src/main/java/org/openhab/binding/zwave/handler/ZWaveSerialHandler.java

```diff
@@ -80,49 +89,86 @@ public void initialize() {
+
+    }
+
+    /**
+     * Gets a serial port identifier for a given name.
+     * Workaround for getIdentifier in SerialPortManager class, because it doesn't detected correctly that the device is
+     * unplugged.
+     *
+     * @param the name
+     * @return a serial port identifier or null
+     */
+    private @Nullable SerialPortIdentifier getSerialPortIdentifier(final String name) {
+        Optional<SerialPortIdentifier> opt = serialPortManager.getIdentifiers().filter(id -> id.getName().equals(name))
+                .findFirst();
+        if (opt.isPresent()) {
+            return opt.get();
+        } else {
+            return null;
+        }
+    }
+
+    private void watchSerialPort() {
         try {
-            SerialPortIdentifier portIdentifier = serialPortManager.getIdentifier(portId);
+            SerialPortIdentifier portIdentifier = getSerialPortIdentifier(portId);
             if (portIdentifier == null) {
+                if (serialPort != null) {
+                    onSerialPortError(ZWaveBindingConstants.OFFLINE_SERIAL_NOTFOUND);
+                }
                 updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.OFFLINE.COMMUNICATION_ERROR,
                         ZWaveBindingConstants.OFFLINE_SERIAL_NOTFOUND + " [\"" + portId + "\"]");
                 return;
             }
+
+            if (serialPort != null) {
+                return;
+            }
+
+            logger.info("Connecting to serial port '{}'", portId);
             SerialPort commPort = portIdentifier.open("org.openhab.binding.zwave", 2000);
             serialPort = commPort;
-            serialPort.setSerialPortParams(115200, SerialPort.DATABITS_8, SerialPort.STOPBITS_1,
-                    SerialPort.PARITY_NONE);
-            serialPort.enableReceiveThreshold(1);
-            serialPort.enableReceiveTimeout(SERIAL_RECEIVE_TIMEOUT);
+            commPort.setSerialPortParams(115200, SerialPort.DATABITS_8, SerialPort.STOPBITS_1, SerialPort.PARITY_NONE);
+            commPort.enableReceiveThreshold(1);
+            commPort.enableReceiveTimeout(SERIAL_RECEIVE_TIMEOUT);
+            inputStream = commPort.getInputStream();
+            outputStream = commPort.getOutputStream();
             logger.debug("Starting receive thread");
-            receiveThread = new ZWaveReceiveThread();
-            receiveThread.start();
+
+            ZWaveReceiveThread zWaveReceiveThread = new ZWaveReceiveThread();
+            receiveThread = zWaveReceiveThread;
+            zWaveReceiveThread.start();
 
             // RXTX serial port library causes high CPU load
             // Start event listener, which will just sleep and slow down event loop
-            serialPort.addEventListener(receiveThread);
-            serialPort.notifyOnDataAvailable(true);
+            commPort.addEventListener(zWaveReceiveThread);
+            commPort.notifyOnDataAvailable(false);
 
             logger.info("Serial port is initialized");
```

Comment:
This can probably be debug

Reply:
Ummmm... wouldn't it be better if we log as debug the "```logger.info("Connecting to serial port '{}'", portId);```" message and leave this as it is? IMHO, the "is initialized" message is more meaningful than the "connecting to...". Anyway, as you wish :wink:

Reply:
I don't think we need this message as a general message. We should really just log exceptions (ie things that don't go as expected) at info since the user is really only going to look at the log when there is a problem (IMHO).